pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -491,14 +496,59 @@ func shouldIgnore(a admission.Attributes) (bool, error) {
 	if pod.Spec.OS != nil && pod.Spec.OS.Name == coreapi.Windows {
 		return true, nil
 	}
-	// if this is an update, see if we are only updating the ownerRef.  Garbage collection does this
+
+	if a.GetOperation() == admission.Update {
+		oldPod, ok := a.GetOldObject().(*coreapi.Pod)
+		if !ok {
+			return false, admission.NewForbidden(a, fmt.Errorf("object was marked as kind pod but was unable to be converted: %v", a.GetOldObject()))
+		}
+
+		// never ignore any spec changes
+		if !kapihelper.Semantic.DeepEqual(pod.Spec, oldPod.Spec) {
+			return false, nil
+		}
+
+		// see if we are only doing meta changes that should be ignored during admission
+		// for example, the OVN controller adds informative networking annotations that shouldn't cause the pod to go through admission again
+		if shouldIgnoreMetaChanges(pod, oldPod) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func shouldIgnoreMetaChanges(newPod, oldPod *coreapi.Pod) bool {
+	// see if we are only updating the ownerRef. Garbage collection does this
 	// and we should allow it in general, since you had the power to update and the power to delete.
 	// The worst that happens is that you delete something, but you aren't controlling the privileged object itself
-	if a.GetOperation() == admission.Update && rbacregistry.IsOnlyMutatingGCFields(a.GetObject(), a.GetOldObject(), kapihelper.Semantic) {
-		return true, nil
+	if rbacregistry.IsOnlyMutatingGCFields(newPod, oldPod, kapihelper.Semantic) {
+		return true
 	}
 
-	return false, nil
+	// check if we're adding or changing only annotations from the ignore list
+	for key, newVal := range newPod.ObjectMeta.Annotations {
+		if oldVal, ok := oldPod.ObjectMeta.Annotations[key]; ok && newVal == oldVal {
+			continue
+		}
+
+		if !ignoredAnnotations.Has(key) {
+			return false
+		}
+	}
+
+	// check if we're removing only annotations from the ignore list
+	for key := range oldPod.ObjectMeta.Annotations {
+		if _, ok := newPod.ObjectMeta.Annotations[key]; ok {
+			continue
+		}
+
+		if !ignoredAnnotations.Has(key) {
+			return false
+		}
+	}
+
+	return true
```

Comment:
This looks incorrect.  If I change a label (not checked above), I would expect shouldIgnoreMetaChanges to be false.

While you have an argument for ignoring fields not inspected by SCC, this PR appears to be working on the assumption of "default check" versus "default don't check".

If you intended for label mutation to be checked as it was before, you're short a test.